graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/RestHighLevelClientProvider.java

```diff
@@ -64,22 +61,24 @@ public RestHighLevelClientProvider(
             @Named("elasticsearch_discovery_default_scheme") String defaultSchemeForDiscoveredNodes,
             @Named("elasticsearch_use_expect_continue") boolean useExpectContinue,
             CredentialsProvider credentialsProvider) {
-        client = buildClient(
-                hosts,
-                connectTimeout,
-                socketTimeout,
-                maxTotalConnections,
-                maxTotalConnectionsPerRoute,
-                useExpectContinue,
-                credentialsProvider);
+        clientProvider = Suppliers.memoize(() -> {
+            final RestHighLevelClient client = buildClient(hosts,
+                    connectTimeout,
+                    socketTimeout,
+                    maxTotalConnections,
+                    maxTotalConnectionsPerRoute,
+                    useExpectContinue,
+                    credentialsProvider);
 
-        sniffer = discoveryEnabled
-                ? createNodeDiscoverySniffer(client.getLowLevelClient(), discoveryFrequency, defaultSchemeForDiscoveredNodes, discoveryFilter)
-                : null;
+            synchronized (this) {
+                if (sniffer != null && discoveryEnabled) {
```

Comment:
Without a test: is sniffer != null correct here? I'd think it should be sniffer == null.

Reply:
:white_check_mark: